src/Core/Compiler/Generator/StatementGenerator.cs

```diff
@@ -125,6 +125,29 @@ internal static class StatementGenerator {
                 writer.Indent--;
                 writer.WriteLine("}");
             }
+            else if (statement.CollectionExpression.EvaluatedType.IsArray)
+            {
+                string dataSourceVariableName = statement.LoopVariable.GeneratedName;
+                string indexVariableName = dataSourceVariableName + "_index";
+
+                // var $$ = ({CollectionExpression});
+                writer.Write("var " + dataSourceVariableName + " = (");
```

Comment:
Why do we open and close brackets around the collection we are using?

Reply:
it's a good practice to inline expressions inside parenthesis to ensure that they don't affect the outer expression

Reply:
Oh really, learn something new every day. Where did you read up about that? Love to have a read myself :)

Reply:
I honestly have no idea where to find documentation about this.
I guess it's something you learn after fixing bugs caused by this in the past 😆